xbmc/addons/AddonManager.cpp

```diff
@@ -984,242 +671,12 @@ bool CAddonMgr::IsSystemAddon(const std::string& id)
-    return true;
-
-  std::vector<std::string> supportedPlatforms = {
-    "all",
-#if defined(TARGET_ANDROID)
-    "android",
-#if defined(__ARM_ARCH_7A__)
-    "android-armv7",
-#elif defined(__aarch64__)
-    "android-aarch64",
-#elif defined(__i686__)
-    "android-i686",
-#else
-    #warning no architecture dependant platform tag
-#endif
-#elif defined(TARGET_FREEBSD)
-    "freebsd",
-    "linux",
-#elif defined(TARGET_LINUX)
-    "linux",
-#elif defined(TARGET_WINDOWS_DESKTOP)
-    "windx",
-    "windows",
-#if defined(_M_IX86)
-    "windows-i686",
-#elif defined(_M_AMD64)
-    "windows-x86_64",
-#else
-#error no architecture dependant platform tag
-#endif
-#elif defined(TARGET_WINDOWS_STORE)
-    "windowsstore",
-#elif defined(TARGET_DARWIN_IOS)
-    "ios",
-#if defined(__ARM_ARCH_7A__)
-    "ios-armv7",
-#elif defined(__aarch64__)
-    "ios-aarch64",
-#else
-#warning no architecture dependant platform tag
-#endif
-#elif defined(TARGET_DARWIN_OSX)
-    "osx",
-#if defined(__x86_64__)
-    "osx64",
-    "osx-x86_64",
-#elif defined(__i686__)
-    "osx-i686",
-    "osx32",
-#else
-#warning no architecture dependant platform tag
-#endif
-#endif
-  };
-
-  return std::find_first_of(platforms.begin(), platforms.end(),
-      supportedPlatforms.begin(), supportedPlatforms.end()) != platforms.end();
-}
-
-cp_cfg_element_t *CAddonMgr::GetExtElement(cp_cfg_element_t *base, const char *path)
-{
-  cp_cfg_element_t *element = NULL;
-  if (base)
-    element = cp_lookup_cfg_element(base, path);
-  return element;
-}
-
-bool CAddonMgr::GetExtElements(cp_cfg_element_t *base, const char *path, ELEMENTS &elements)
-{
-  if (!base || !path)
-    return false;
-
-  for (unsigned int i = 0; i < base->num_children; i++)
-  {
-    std::string temp = base->children[i].name;
-    if (!temp.compare(path))
-      elements.push_back(&base->children[i]);
-  }
-
-  return !elements.empty();
-}
-
-const cp_extension_t *CAddonMgr::GetExtension(const cp_plugin_info_t *props, const char *extension) const
-{
-  if (!props)
-    return NULL;
-  for (unsigned int i = 0; i < props->num_extensions; ++i)
-  {
-    if (0 == strcmp(props->extensions[i].ext_point_id, extension))
-      return &props->extensions[i];
-  }
-  return NULL;
-}
-
-std::string CAddonMgr::GetExtValue(cp_cfg_element_t *base, const char *path) const
-{
-  const char *value = "";
-  if (base && (value = cp_lookup_cfg_value(base, path)))
-    return value;
-  else
-    return "";
-}
-
-bool CAddonMgr::GetExtList(cp_cfg_element_t *base, const char *path, std::vector<std::string> &result) const
-{
-  result.clear();
-  if (!base || !path)
-    return false;
-  const char *all = cp_lookup_cfg_value(base, path);
-  if (!all || *all == 0)
-    return false;
-  StringUtils::Tokenize(all, result, ' ');
-  return true;
-}
-
-std::string CAddonMgr::GetPlatformLibraryName(cp_cfg_element_t *base) const
-{
-  std::string libraryName;
-#if defined(TARGET_ANDROID)
-  libraryName = GetExtValue(base, "@library_android");
```

Comment:
Previously an `addon.xml` could include entries like `library_android=`, `library_windows` to point to different binary addon dlls for different platforms all in the same zip file. It looks like this functionality was removed completely? Was that intentional?

Reply:
Is still possible, this now only on other place now https://github.com/xbmc/xbmc/blob/master/xbmc/addons/addoninfo/AddonInfoBuilder.cpp#L537-L566

And addon.xml can look like, them:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<addon
  id="visualization.shadertoy"
  version="2.0.0"
  name="Shadertoy"
  provider-name="Team Kodi">
  <requires>@ADDON_DEPENDS@</requires>
  <extension
    point="xbmc.player.musicviz"
    library_windows="visualization.shadertoy-win64.dll"
    library_osx="visualization.shadertoy-mac64.dylib"
    library_linux=""visualization.shadertoy-linux64.so"/>"/>
  <extension point="xbmc.addon.metadata">
    <summary lang="en">Visualizations from Shadertoy</summary>
    <description lang="en">This music visualization is based on Shadertoy based GPU programs.</description>
    <platform>windows-x86_64 osx-x86_64 linux</platform> <!-- Here to set manually the supported platforms -->
  </extension>
</addon>
```

About platform visible here https://github.com/xbmc/xbmc/blob/master/xbmc/addons/addoninfo/AddonInfoBuilder.cpp#L568-L628

Only was it a very long time not used with binary addon where have different libs together, at that time Kodi supported only 32bit x86.

Thats why no cpu in `GetPlatformLibraryName` and on xml by `platform` only linux, but also because Linux might not work 100% in other places (for example, built under Ubuntu, used on Fedora), not as easy as Mac or Windows.

Reply:
Aha, thanks for the link. My search failed me because github had collapsed some of the files in the diff.